fccodegen/templates/wl/weblab_model.pyx

```diff
@@ -1,47 +1,50 @@
-
 # cython: profile=True
-## @file
-##
-## This source file was generated from CellML.
-##
-## Model: tentusscher_model_2004_epi
-##
-## Processed by pycml - CellML Tools in Python
-##     (translators: , pycml: , optimize: )
-## on Fri May 18 13:29:33 2018
-##
-## <autogenerated>
-
-import numpy as np
+#
+#
+# Web Lab model {{name}}
+#
+# Generated by {{generator}} on {{date}}
+#
+#
+#
+cimport fc.sundials.sundials as Sundials
+cimport libc.math as math
+cimport numpy as np
 
 import fc.simulations.model as Model
 import fc.utility.environment as Env
 import fc.language.values as V
-
-cimport libc.math as math
-cimport numpy as np
+from fc.sundials.solver cimport CvodeSolver
+from fc.utility.error_handling import ProtocolError
+import numpy as np
 import os
 import shutil
 import sys
 
-from fc.sundials.solver cimport CvodeSolver
-cimport fc.sundials.sundials as Sundials
-from fc.utility.error_handling import ProtocolError
 
 cdef int _EvaluateRhs(Sundials.realtype var_environment__time, Sundials.N_Vector y, Sundials.N_Vector ydot, void* user_data):
+    """
+    Main model method: evaluates the model's RHS.
+    """
+    #TODO: What's this?
     model = <object>user_data
     cdef np.ndarray[Sundials.realtype, ndim=1] parameters = <np.ndarray>model.parameters
-    # State variables
-    cdef double var_rapid_time_dependent_potassium_current_Xr1_gate__Xr1 = (<Sundials.N_VectorContent_Serial>y.content).data[0]
-    cdef double var_rapid_time_dependent_potassium_current_Xr2_gate__Xr2 = (<Sundials.N_VectorContent_Serial>y.content).data[1]
 
+    # Unpack state variables
+    {%- for state in states %}
+    cdef double {{state[1]}} = (<Sundials.N_VectorContent_Serial>y.content).data[{{state[0]}}]
```

Comment:
It might be nicer to make `states` a list of dictionaries so you can do things like `state.var_name` or `state.index` here.

Reply:
Shall I use a `namedtuple`?

Reply:
You could do I guess, but I was just thinking of filling the `states` list using
```python
        states.append({'index': i, 'var_name': symbol_name(state), 'deriv_name': derivative_name(state)})
```
which I think should do the trick to enable dot-lookup name access.

Reply:
Is that a jinja thing? Will give it a go!